java/vector/src/main/java/org/apache/arrow/vector/NullableBigIntVector.java

```diff
@@ -0,0 +1,318 @@
+      }
+      else {
+         BitVectorHelper.setValidityBit(validityBuffer, index, 0);
+      }
+   }
+
+   /**
+    * Set the element at the given index to the value set in data holder.
+    *
+    * @param index   position of element
+    * @param holder  data holder for value of element
+    */
+   public void set(int index, BigIntHolder holder){
+      BitVectorHelper.setValidityBitToOne(validityBuffer, index);
+      setValue(index, holder.value);
+   }
+
+   /**
+    * Same as {@link #set(int, long)} except that it handles the
+    * case when index is greater than or equal to existing
+    * value capacity {@link #getValueCapacity()}.
+    *
+    * @param index   position of element
+    * @param value   value of element
+    */
+   public void setSafe(int index, long value) {
+      handleSafe(index);
+      set(index, value);
+   }
+
+   /**
+    * Same as {@link #set(int, NullableBigIntHolder)} except that it handles the
+    * case when index is greater than or equal to existing
+    * value capacity {@link #getValueCapacity()}.
+    *
+    * @param index   position of element
+    * @param holder  nullable data holder for value of element
+    */
+   public void setSafe(int index, NullableBigIntHolder holder) throws IllegalArgumentException {
+      handleSafe(index);
+      set(index, holder);
+   }
+
+   /**
+    * Same as {@link #set(int, BigIntHolder)} except that it handles the
+    * case when index is greater than or equal to existing
+    * value capacity {@link #getValueCapacity()}.
+    *
+    * @param index   position of element
+    * @param holder  data holder for value of element
+    */
+   public void setSafe(int index, BigIntHolder holder){
+      handleSafe(index);
+      set(index, holder);
+   }
+
+   /**
+    * Set the element at the given index to null.
+    *
+    * @param index   position of element
+    */
+   public void setNull(int index){
+      handleSafe(index);
+      /* not really needed to set the bit to 0 as long as
+       * the buffer always starts from 0.
+       */
+      BitVectorHelper.setValidityBit(validityBuffer, index, 0);
+   }
+
+   public void set(int index, int isSet, long value) {
+      if (isSet > 0) {
+         set(index, value);
+      } else {
+         BitVectorHelper.setValidityBit(validityBuffer, index, 0);
+      }
+   }
+
+   public void setSafe(int index, int isSet, long value) {
+      handleSafe(index);
+      set(index, isSet, value);
+   }
+
+
+   /******************************************************************
+    *                                                                *
+    *          helper routines currently                             *
+    *          used in JsonFileReader                                *
+    *                                                                *
+    ******************************************************************/
+
+
+   public static ArrowBuf set(ArrowBuf buffer, BufferAllocator allocator,
```

Comment:
Not sure about adding a public API just for the json reader. Is there another way to do this without adding a public API?

Reply:
The obvious another way is to do this in Json code itself. I had that earlier but it made the code very unreadable (since all of this logic was then interspersed into the giant switch block in JsonFileReader/JsonFileWriter code).